### PR TITLE
Refactor attribute parser

### PR DESCRIPTION
**Pull request by ttuegel**
In preparation to add more attributes (`smtlib` and `smt-lemma` next), and in response to remarks by other contributors that the attribute parser is difficult to understand, I simplified the parser and added many test cases.

Besides simplifying the attribute parser, I also changed `StepperAttributes` to distinguish clearly between attributes that are declared in the Kore definition and attributes that are inferred from the declared attributes.

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/src/Kore/IndexedModule/IndexedModule.hs [hunk 1]
```diff
@@ -720,11 +722,12 @@ metaNameForObjectSort name = "#`" ++ name
 
 See also: 'parseAttributes'
```
Comment (traiansf) [inline src/main/haskell/kore/src/Kore/IndexedModule/IndexedModule.hs]: Qualify `parseAttributes` in the comment?

**Comment by vladciobanu** [comment]
I'm not a big fan of `Default`, but I realize it was already there for `Attributes`. If we all agree, we should create a trello card to remove it.

**Comment by ttuegel** [comment]
@traiansf The lenses are necessary on some level to make the attribute parsers [composable](https://github.com/kframework/kore/pull/326/files#diff-670ed0a1188c6486add14ac9bd0951e2R77), but I can change it so that the lenses do not need to be the "default" way to access `StepperAttributes`.

**Comment by ttuegel** [comment]
> If we all agree, we should create a trello card to remove it.

@vladciobanu I would like to remove it too, but I'm not sure what it should be replaced with. Every attribute must have a default value in case it is omitted.

**Pull request merged**